### PR TITLE
Refine MilvusDataStore/ZillizDataStore

### PR DESCRIPTION
**Pull request by yhmo**
Major changes:
1. Change the collection schema to use DocumentChunk.id as the primary key
2. Use cosine similarity instead of euclidean distance
3. ZillizDataStore derives from MilvusDataStore to avoid duplicated code
4. Add new env variables "MILVUS_INDEX_PARAMS" and "MILVUS_SEARCH_PARAMS" to overwrite default index/search parameters
5. Add new env variables "MILVUS_CONSISTENCY_LEVEL" and "ZILLIZ_CONSISTENCY_LEVEL" to overwrite default data consistency level.
6. Update test cases accordingly

### CODE REVIEW

**Review by filip-halt — APPROVED** [review]
lgtm

File: datastore/providers/milvus_datastore.py
Comment (filip-halt) [inline datastore/providers/milvus_datastore.py:106]: consistency_level is missing from the ZillizDatastore init, causing errors in creation.

Comment (yhmo) [reply datastore/providers/milvus_datastore.py:106]: I have added a line "self._consistency_level = "Bounded"" in the ZillizDatastore.__init__

**Review by isafulf — COMMENTED** [review]
Note: passes tests

lgtm -- after you make the README change I'll merge, thanks Filip.

**Comment by yhmo** [comment]
Hi @isafulf ,
I have rebased the pull request and solve the readme conflict. Please help me merge it.
Thanks!

**Review by isafulf — APPROVED** [review]
Thanks for the updates, merging!

**Pull request merged**
